pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -234,6 +419,7 @@ func (p *Propagator) processEvent() {
 			return
 		}
 		p.removeNode(existingNode)
+		// this read of "dependents" doesn't require a RLock, because it's in processEvent(), which is single-threaded and is the only writer of "dependents".
```

Comment:
"node" is a private and its methods are private, too. I didn't modify a node in the unit tests either.

Anyway, adding a lock doesn't cost much, I'll add it.